modules/k8s_eks_addons/load-balancer-controller.tf

```
@@ -0,0 +1,424 @@
+  policy      = data.aws_iam_policy_document.aws_load_balancer_controller[0].json
+  tags        = var.addon_context.tags
+}
+
+resource "kubernetes_service_account_v1" "aws_load_balancer_controller" {
+  count = var.aws_load_balancer_controller_config.enable ? 1 : 0
+  metadata {
+    name        = local.aws_load_balancer_controller_service_account
+    namespace   = local.aws_load_balancer_controller_namespace
+    annotations = { "eks.amazonaws.com/role-arn" : aws_iam_role.aws_load_balancer_controller[0].arn }
+  }
+  automount_service_account_token = true
+}
+
+resource "aws_iam_role" "aws_load_balancer_controller" {
+  count = var.aws_load_balancer_controller_config.enable ? 1 : 0
+
+  name        = format("%s-%s-%s", var.addon_context.eks_cluster_id, trim(local.aws_load_balancer_controller_service_account, "-*"), "irsa")
+  description = "AWS IAM Role for the Kubernetes service account ${local.aws_load_balancer_controller_service_account}."
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Federated" : "arn:${var.addon_context.aws_partition_id}:iam::${var.addon_context.aws_caller_identity_account_id}:oidc-provider/${var.addon_context.eks_oidc_issuer_url}"
+        },
+        "Action" : "sts:AssumeRoleWithWebIdentity",
+        "Condition" : {
+          "StringLike" : {
+            "${var.addon_context.eks_oidc_issuer_url}:sub" : "system:serviceaccount:${local.aws_load_balancer_controller_namespace}:${local.aws_load_balancer_controller_service_account}",
+            "${var.addon_context.eks_oidc_issuer_url}:aud" : "sts.amazonaws.com"
+          }
+        }
+      }
+    ]
+  })
+  force_detach_policies = true
+
+  tags = var.addon_context.tags
+}
+
+resource "aws_iam_role_policy_attachment" "aws_load_balancer_controller" {
+  count = var.aws_load_balancer_controller_config.enable ? 1 : 0
+
+  policy_arn = aws_iam_policy.aws_load_balancer_controller[0].arn
+  role       = aws_iam_role.aws_load_balancer_controller[0].name
+}
+
+
+resource "helm_release" "aws_load_balancer_controller" {
+  count = var.aws_load_balancer_controller_config.enable ? 1 : 0
+
+  name       = local.aws_load_balancer_controller_name
+  namespace  = local.aws_load_balancer_controller_namespace
+  repository = var.aws_load_balancer_controller_config.helm_repository
+  chart      = local.aws_load_balancer_controller_name
+  version    = var.aws_load_balancer_controller_config.helm_version
+  timeout    = 1200
+  values = [templatefile("${path.module}/templates/load_balancer_controller_values.yaml", {
+    aws_region      = var.addon_context.aws_region_name
+    eks_cluster_id  = var.addon_context.eks_cluster_id
+    repository      = "${local.amazon_container_image_registry_uris[var.addon_context.aws_region_name]}/amazon/aws-load-balancer-controller"
+    service_account = local.aws_load_balancer_controller_service_account
+    }),
```

We should pass the customer values to the helm release:
```suggestion
    }),
    var.aws_load_balancer_controller_config.chart_values
```

---

This has now been added.